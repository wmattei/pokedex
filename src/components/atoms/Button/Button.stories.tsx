import { action } from '@storybook/addon-actions';
import { boolean, text, withKnobs } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import * as React from 'react';
import Button from '.';
import CenterView from '../../../../storybook/stories/CenterView';

storiesOf('Button', module)
  .addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
  .addDecorator(withKnobs)
  .add('Button', () => (
    <Button
      onPress={action('clicked-text')}
      loading={boolean('Loading', false)}
      inverted={boolean('Inverted', false)}
      disabled={boolean('Disabled', false)}
      title={text('Title', 'Hello World')}
    />
  ))
  .add('Emoji', () => (
    <Button onPress={action('clicked-text')} title={text('Emoji', 'ðŸ˜•')} />
  ))
  .add('Em estado de loading', () => (
    <Button
      onPress={action('clicked-text')}
      title={text('Title', 'Hello World')}
      loading
    />
  ))
  .add('BotÃ£o invertido', () => (
    <Button
      onPress={action('clicked-text')}
      title={text('Title', 'Hello World')}
      inverted
    />
  ))
  .add('BotÃ£o disabled', () => (
    <Button
      title={text('Title', 'Hello World')}
      onPress={action('clicked-text')}
      disabled
    />
  ))
  .add('BotÃ£o invertido disabled', () => (
    <Button
      title={text('Title', 'Hello World')}
      onPress={action('clicked-text')}
      disabled
      inverted
    />
  ));
