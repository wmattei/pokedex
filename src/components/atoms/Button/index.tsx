import * as React from 'react';
import { Button as PaperButton } from 'react-native-paper';

type Props = {
  onPress: () => void;
  title?: string;
  disabled?: boolean;
  inverted?: boolean;
  loading?: boolean;
  children: React.ReactNode;
};

export default function Button({
  onPress,
  disabled,
  title,
  children,
  inverted,
  loading,
}: Props) {
  return (
    <PaperButton
      loading={loading}
      mode={inverted ? 'outlined' : 'contained'}
      accessibilityLabel={title}
      onPress={onPress}
      disabled={disabled}
    >
      {children}
    </PaperButton>
  );
}
