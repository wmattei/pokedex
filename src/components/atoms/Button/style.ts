import { StyleSheet } from 'react-native';

type Props = {
  inverted?: boolean;
};

export const style = ({ inverted }: Props) =>
  StyleSheet.create({
    disabledButton: {
      backgroundColor: !inverted ? '#dddde1' : 'white',
    },
    disabledText: {
      color: inverted ? '#dddde1' : 'gray',
    },
  });
