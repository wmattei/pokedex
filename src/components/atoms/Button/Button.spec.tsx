import * as React from 'react';
import { create } from 'react-test-renderer';
import Button from '.';

describe('Button', () => {
  describe('Normal button', () => {
    it('Renders correctly', () => {
      const tree = create(
        <Button onPress={() => console.log('Pressed')} title="Hello world" />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('Renders correctly loading state', () => {
      const tree = create(
        <Button
          loading
          onPress={() => console.log('Pressed')}
          title="Hello world"
        />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('Renders correctly disabled state', () => {
      const tree = create(
        <Button
          disabled
          onPress={() => console.log('Pressed')}
          title="Hello world"
        />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('Inverted button', () => {
    it('Renders correctly', () => {
      const tree = create(
        <Button
          inverted
          onPress={() => console.log('Pressed')}
          title="Hello world"
        />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('Renders correctly loading state', () => {
      const tree = create(
        <Button
          loading
          inverted
          onPress={() => console.log('Pressed')}
          title="Hello world"
        />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('Renders correctly disabled state', () => {
      const tree = create(
        <Button
          disabled
          inverted
          onPress={() => console.log('Pressed')}
          title="Hello world"
        />
      ).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
