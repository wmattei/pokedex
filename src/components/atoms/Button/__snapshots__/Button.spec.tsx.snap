// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Inverted button Renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "borderColor": "rgba(0, 0, 0, 0.29)",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "elevation": 0,
      "minWidth": 64,
    }
  }
>
  <View
    accessibilityLabel="Hello world"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "letterSpacing": 1,
                "marginHorizontal": 16,
                "marginVertical": 9,
                "textAlign": "center",
              },
              undefined,
              Object {
                "textTransform": "uppercase",
              },
              Object {
                "color": "#6200ee",
                "fontFamily": "System",
                "fontWeight": "500",
              },
              Object {
                "fontFamily": "System",
                "fontWeight": "500",
              },
              undefined,
            ],
          ]
        }
      >
        <Text>
          Hello world
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`Button Inverted button Renders correctly disabled state 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "rgba(0, 0, 0, 0.29)",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "elevation": 0,
      "minWidth": 64,
    }
  }
>
  <View
    accessibilityLabel="Hello world"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "letterSpacing": 1,
                "marginHorizontal": 16,
                "marginVertical": 9,
                "textAlign": "center",
              },
              undefined,
              Object {
                "textTransform": "uppercase",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.32)",
                "fontFamily": "System",
                "fontWeight": "500",
              },
              Object {
                "fontFamily": "System",
                "fontWeight": "500",
              },
              undefined,
            ],
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "#dddde1",
            }
          }
        >
          Hello world
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`Button Inverted button Renders correctly loading state 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "borderColor": "rgba(0, 0, 0, 0.29)",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 0.5,
      "elevation": 0,
      "minWidth": 64,
    }
  }
>
  <View
    accessibilityLabel="Hello world"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityRole="progressbar"
        accessible={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "marginLeft": 12,
              "marginRight": -4,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "height": 16,
              "opacity": 1,
              "width": 16,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 16,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 16,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 8,
                    "overflow": "hidden",
                    "width": 16,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": 16,
                      "transform": Array [
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "rotate": "-165deg",
                        },
                      ],
                      "width": 16,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 8,
                        "overflow": "hidden",
                        "width": 16,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "borderColor": "#6200ee",
                          "borderRadius": 8,
                          "borderWidth": 1.6,
                          "height": 16,
                          "width": 16,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 16,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 16,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 8,
                    "overflow": "hidden",
                    "top": 8,
                    "width": 16,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": 16,
                      "transform": Array [
                        Object {
                          "translateY": -8,
                        },
                        Object {
                          "rotate": "345deg",
                        },
                      ],
                      "width": 16,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 8,
                        "overflow": "hidden",
                        "width": 16,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "borderColor": "#6200ee",
                          "borderRadius": 8,
                          "borderWidth": 1.6,
                          "height": 16,
                          "width": 16,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "letterSpacing": 1,
                "marginHorizontal": 16,
                "marginVertical": 9,
                "textAlign": "center",
              },
              undefined,
              Object {
                "textTransform": "uppercase",
              },
              Object {
                "color": "#6200ee",
                "fontFamily": "System",
                "fontWeight": "500",
              },
              Object {
                "fontFamily": "System",
                "fontWeight": "500",
              },
              undefined,
            ],
          ]
        }
      >
        <Text>
          Hello world
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`Button Normal button Renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#6200ee",
      "borderColor": "transparent",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 0,
      "elevation": 2,
      "minWidth": 64,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 0.75,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 1.5,
    }
  }
>
  <View
    accessibilityLabel="Hello world"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "letterSpacing": 1,
                "marginHorizontal": 16,
                "marginVertical": 9,
                "textAlign": "center",
              },
              undefined,
              Object {
                "textTransform": "uppercase",
              },
              Object {
                "color": "#ffffff",
                "fontFamily": "System",
                "fontWeight": "500",
              },
              Object {
                "fontFamily": "System",
                "fontWeight": "500",
              },
              undefined,
            ],
          ]
        }
      >
        <Text>
          Hello world
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`Button Normal button Renders correctly disabled state 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#dddde1",
      "borderColor": "transparent",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 0,
      "elevation": 0,
      "minWidth": 64,
    }
  }
>
  <View
    accessibilityLabel="Hello world"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "letterSpacing": 1,
                "marginHorizontal": 16,
                "marginVertical": 9,
                "textAlign": "center",
              },
              undefined,
              Object {
                "textTransform": "uppercase",
              },
              Object {
                "color": "rgba(0, 0, 0, 0.32)",
                "fontFamily": "System",
                "fontWeight": "500",
              },
              Object {
                "fontFamily": "System",
                "fontWeight": "500",
              },
              undefined,
            ],
          ]
        }
      >
        <Text
          style={
            Object {
              "color": "gray",
            }
          }
        >
          Hello world
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`Button Normal button Renders correctly loading state 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#6200ee",
      "borderColor": "transparent",
      "borderRadius": 4,
      "borderStyle": "solid",
      "borderWidth": 0,
      "elevation": 2,
      "minWidth": 64,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 0.75,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 1.5,
    }
  }
>
  <View
    accessibilityLabel="Hello world"
    accessibilityRole="button"
    accessibilityState={
      Object {
        "disabled": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "overflow": "hidden",
        },
        Object {
          "borderRadius": 4,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <View
        accessibilityRole="progressbar"
        accessible={true}
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "marginLeft": 12,
              "marginRight": -4,
            },
          ]
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "height": 16,
              "opacity": 1,
              "width": 16,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 16,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 16,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 8,
                    "overflow": "hidden",
                    "width": 16,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": 16,
                      "transform": Array [
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "rotate": "-165deg",
                        },
                      ],
                      "width": 16,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 8,
                        "overflow": "hidden",
                        "width": 16,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "borderColor": "#ffffff",
                          "borderRadius": 8,
                          "borderWidth": 1.6,
                          "height": 16,
                          "width": 16,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "bottom": 0,
                "justifyContent": "center",
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "height": 16,
                  "transform": Array [
                    Object {
                      "rotate": "45deg",
                    },
                  ],
                  "width": 16,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "height": 8,
                    "overflow": "hidden",
                    "top": 8,
                    "width": 16,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "height": 16,
                      "transform": Array [
                        Object {
                          "translateY": -8,
                        },
                        Object {
                          "rotate": "345deg",
                        },
                      ],
                      "width": 16,
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "height": 8,
                        "overflow": "hidden",
                        "width": 16,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "borderColor": "#ffffff",
                          "borderRadius": 8,
                          "borderWidth": 1.6,
                          "height": 16,
                          "width": 16,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Array [
              Object {
                "letterSpacing": 1,
                "marginHorizontal": 16,
                "marginVertical": 9,
                "textAlign": "center",
              },
              undefined,
              Object {
                "textTransform": "uppercase",
              },
              Object {
                "color": "#ffffff",
                "fontFamily": "System",
                "fontWeight": "500",
              },
              Object {
                "fontFamily": "System",
                "fontWeight": "500",
              },
              undefined,
            ],
          ]
        }
      >
        <Text>
          Hello world
        </Text>
      </Text>
    </View>
  </View>
</View>
`;
